### ISSUE TITLE
initHeapQueue fails.  Should it be marked as deprecated in the docs?

### ISSUE THREAD

**Issue by stephenwithav** [nim-lang/Nim#11241]
The documentation for the `heapqueue` package includes failing `initHeapQueue[int]()` tests.

The code fails because `initHeapQueue`'s body is merely `discard`.

Should the documentation be updated to reflect working code?

Failing example:

```nim
import heapqueue

var heap = initHeapQueue[int]()
heap.push(8)
heap.push(2)
heap.push(5)
# The first element is the lowest element
assert heap[0] == 2
# Remove and return the lowest element
assert heap.pop() == 2
# The lowest element remaining is 5
assert heap[0] == 5
```

Working example:

```nim
import heapqueue

var heap: HeapQueue[int]
heap.push(8)
heap.push(2)
heap.push(5)
# The first element is the lowest element
assert heap[0] == 2
# Remove and return the lowest element
assert heap.pop() == 2
# The lowest element remaining is 5
assert heap[0] == 5
```

Should the `{.deprecated.}` pragma be added to `initHeapQueue`?

Nim: 0.19.4

**Comment by stephenwithav** [comment]
The [source](https://github.com/nim-lang/Nim/blob/9402c82e803d133e0b845a7c5c79c261781e7d8d/lib/pure/collections/heapqueue.nim) looks like it needs a good bit of cleaning up related to `initHeapQueue`.  Mind if I tackle the issue?

**Comment by GULPF** [comment]
The first example works on devel. `initHeapQueue` doesn't exist on 0.19.4, but for some reason the stable docs are built from devel which is why it's included in the docs.

**Comment by stephenwithav** [comment]
Thank you for the clarification.  Close this issue?

**Comment by mratsim** [comment]
Seems like a documentation bug that happens from time to time when doc generation does not happen in sync with release. We had this with `openFileStream` in the past https://github.com/nim-lang/Nim/issues/9063.

**Comment by narimiran** [comment]
> Should the `{.deprecated.}` pragma be added to `initHeapQueue`?

It is the other way around.

In devel (and upcoming v0.20), `newHeapQueue` is deprecated (see [here](https://nim-lang.github.io/Nim/heapqueue.html#newHeapQueue)), and `initHeapQueue` is the one you should use.
In stable (v0.19.4 and just released v0.19.6) there is no `initHeapQueue` (see 0.19.6 heapqueue docs [here](https://nim-lang.org/docs/heapqueue.html)).

What I think it happens that you were looking at devel docs, as that is where our website takes you (because devel has vastly improved documentation and in 98% of time you're better of by looking at it than in the stable docs) and in some rare cases there are some changes like this one where you cannot follow devel docs with the stable version of Nim.

This will be fixed once we release v0.20 — the website will take you to the stable docs (which will have all the improvements mentioned above), and people on devel version will have a separate link for devel docs.

Sorry for the confusion. If you would prefer to use stable documentation, you can follow this link: https://nim-lang.org/docs/lib.html
